packages/flutter/lib/src/widgets/overflow_bar.dart

```diff
@@ -0,0 +1,482 @@
+    properties.add(EnumProperty<OverflowBarAlignment>('overflowAlignment', overflowAlignment, defaultValue: OverflowBarAlignment.start));
+    properties.add(EnumProperty<VerticalDirection>('overflowDirection', overflowDirection, defaultValue: VerticalDirection.down));
+    properties.add(EnumProperty<TextDirection>('textDirection', textDirection, defaultValue: null));
+  }
+}
+
+class _OverflowBarParentData extends ContainerBoxParentData<RenderBox> { }
+
+class _RenderOverflowBar extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, _OverflowBarParentData>,
+         RenderBoxContainerDefaultsMixin<RenderBox, _OverflowBarParentData> {
+  _RenderOverflowBar({
+    List<RenderBox> children,
+    double spacing = 0.0,
+    double overflowSpacing = 0.0,
+    OverflowBarAlignment overflowAlignment = OverflowBarAlignment.start,
+    VerticalDirection overflowDirection = VerticalDirection.down,
+    TextDirection textDirection,
+    Clip clipBehavior = Clip.none,
+  }) : assert(spacing != null),
+       assert(overflowSpacing != null),
+       assert(overflowAlignment != null),
+       assert(textDirection != null),
+       assert(clipBehavior != null),
+       _spacing = spacing,
+       _overflowSpacing = overflowSpacing,
+       _overflowAlignment = overflowAlignment,
+       _overflowDirection = overflowDirection,
+       _textDirection = textDirection,
+       _clipBehavior = clipBehavior {
+    addAll(children);
+  }
+
+  double get spacing => _spacing;
+  double _spacing;
+  set spacing (double value) {
+    assert(value != null);
+    if (_spacing == value)
+      return;
+    _spacing = value;
+    markNeedsLayout();
+  }
+
+  double get overflowSpacing => _overflowSpacing;
+  double _overflowSpacing;
+  set overflowSpacing (double value) {
+    assert(value != null);
+    if (_overflowSpacing == value)
+      return;
+    _overflowSpacing = value;
+    markNeedsLayout();
+  }
+
+  OverflowBarAlignment get overflowAlignment => _overflowAlignment;
+  OverflowBarAlignment _overflowAlignment;
+  set overflowAlignment (OverflowBarAlignment value) {
+    assert(value != null);
+    if (_overflowAlignment == value)
+      return;
+    _overflowAlignment = value;
+    markNeedsLayout();
+  }
+
+  VerticalDirection get overflowDirection => _overflowDirection;
+  VerticalDirection _overflowDirection;
+  set overflowDirection (VerticalDirection value) {
+    assert(value != null);
+    if (_overflowDirection == value)
+      return;
+    _overflowDirection = value;
+    markNeedsLayout();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection != value) {
+      _textDirection = value;
+      markNeedsLayout();
+    }
+  }
+
+  Clip get clipBehavior => _clipBehavior;
+  Clip _clipBehavior = Clip.none;
+  set clipBehavior(Clip value) {
+    assert(value != null);
+    if (value != _clipBehavior) {
```

Comment:
The above setters follow the pattern of returning early if its the same, should this one too?

Reply:
👍